ts/client/src/client.ts

```diff
@@ -812,21 +797,105 @@ export class MangoClient {
   }
 
   public async getMangoAccount(
-    mangoAccount: MangoAccount | PublicKey,
+    mangoAccountPk: PublicKey,
     loadSerum3Oo = false,
   ): Promise<MangoAccount> {
-    const mangoAccountPk =
-      mangoAccount instanceof MangoAccount
-        ? mangoAccount.publicKey
-        : mangoAccount;
-    const mangoAccount_ = MangoAccount.from(
+    const mangoAccount = await this.getMangoAccountFromPk(mangoAccountPk);
+    if (loadSerum3Oo) {
+      await mangoAccount?.reloadSerum3OpenOrders(this);
+    }
+    return mangoAccount;
+  }
+
+  private async getMangoAccountFromPk(
+    mangoAccountPk: PublicKey,
+  ): Promise<MangoAccount> {
+    return this.getMangoAccountFromAi(
       mangoAccountPk,
-      await this.program.account.mangoAccount.fetch(mangoAccountPk),
+      (await this.program.provider.connection.getAccountInfo(
+        mangoAccountPk,
+      )) as AccountInfo<Buffer>,
     );
-    if (loadSerum3Oo) {
-      await mangoAccount_?.reloadSerum3OpenOrders(this);
+  }
+
+  private getMangoAccountFromAi(
+    mangoAccountPk: PublicKey,
+    ai: AccountInfo<Buffer>,
+  ): MangoAccount {
+    const decodedMangoAccount = this.program.coder.accounts.decode(
+      'mangoAccount',
+      ai.data,
+    );
+
+    // Start reading tokenConditionalSwaps from this point in account buffer
+    const start =
+      // anchor headers
+      8 +
```

Comment:
I don't like this, and was going to suggest using `coder.size()`, but when I checked that code, it looked like it doesn't actually work with accounts that contain Vecs (assumes their size is 1 byte!).

Maybe re-encode the account and use that size?

How about we make a MangoAccountV2 type that you could try decoding the data as?